src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -330,41 +380,42 @@ private static void validateFieldType(MappedFieldType fieldType) {
         }
     }
 
-    private static void validateQueryTokens(Map<String, Float> queryTokens) {
-        if (null == queryTokens) {
-            throw new IllegalArgumentException("Query tokens cannot be null.");
-        }
-        for (Map.Entry<String, Float> entry : queryTokens.entrySet()) {
-            if (entry.getValue() <= 0) {
-                throw new IllegalArgumentException(
-                    "Feature weight must be larger than 0, feature [" + entry.getValue() + "] has negative weight."
-                );
-            }
-        }
-    }
-
     @Override
     protected boolean doEquals(NeuralSparseQueryBuilder obj) {
         if (this == obj) return true;
         if (Objects.isNull(obj) || getClass() != obj.getClass()) return false;
         if (Objects.isNull(queryTokensSupplier) && Objects.nonNull(obj.queryTokensSupplier)) return false;
         if (Objects.nonNull(queryTokensSupplier) && Objects.isNull(obj.queryTokensSupplier)) return false;
+        if (Objects.nonNull(twoPhaseSharedQueryToken) && Objects.isNull(obj.twoPhaseSharedQueryToken)) return false;
```

Comment:
We don't need to check null here. Line 401 is enough. EqualsBuilder.append checks the case that one of the object is null

Reply:
thanks, removed.

Reply:
I mean we can directly use this
```
EqualsBuilder equalsBuilder = new EqualsBuilder().append(fieldName, obj.fieldName)
            .append(queryText, obj.queryText)
            .append(modelId, obj.modelId)
            .append(maxTokenScore, obj.maxTokenScore)
            .append(twoPhasePruneRatio, obj.twoPhasePruneRatio)
            .append(twoPhaseSharedQueryToken, obj.twoPhaseSharedQueryToken);
```

Reply:
OK, made change  as this.